### PR TITLE
[WIP] Devicecrafting

### PR DESCRIPTION
**Pull request by Iamgoofball**
smash shit together

get finished device
giphy 1

### CODE REVIEW

File: code/modules/research/devicecrafting/device.dm [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/obj/item/devicecrafting/device/proc/on_attack_hand(mob/user)
+	return
+
+/obj/item/devicecrafting/device/proc/on_attack_self(mob/user)
+	return
+
+/obj/item/devicecrafting/device/proc/on_attack_ai(mob/user)
+	return
+
+/obj/item/devicecrafting/device/trigger/tesla // tiles surrouding my tile
```
Comment (optimumtact) [inline code/modules/research/devicecrafting/device.dm]: Funny but lose the comments

File: code/modules/research/devicecrafting/obj_stuff.dm [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/obj
```
Comment (optimumtact) [inline code/modules/research/devicecrafting/obj_stuff.dm]: why does every single object need to hold devices for crafting?

They're just creating the final device in a frame? why is this so high in the typepath?

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -327,6 +327,29 @@ var/global/image/fire_overlay = image("icon" = 'icons/effects/fire.dmi', "icon_s
...
+			if(devicecrafting_holder.devices.len)
+				user << "You remove all the devices from [src]."
+				for(var/obj/item/devicecrafting/device/DE in devicecrafting_holder.devices)
+					DE.loc = get_turf(src)
+					devicecrafting_holder.devices -= DE
+				return
+			else
+				user << "There are no devices to remove!"
+				return
+		devicecrafting_holder.on_attackby(W,user)
```
Comment (Razharas) [inline code/game/objects/items.dm]: WHy not just call ..()

**Review by optimumtact — CHANGES_REQUESTED** [review]

File: code/modules/research/devicecrafting/obj_stuff.dm (hunk 2 above)
Comment (optimumtact) [inline code/modules/research/devicecrafting/obj_stuff.dm]: also this violates the codebase standard

File: tgstation.dme [hunk 4]
```diff
@@ -14,7 +14,7 @@
 
 // BEGIN_INCLUDE
 #include "_maps\__MAP_DEFINES.dm"
-#include "_maps\tgstation2.dm"
+#include "_maps\testmap.dm"
```
Comment (optimumtact) [inline tgstation.dme]: can u not

**Comment by ma44** [comment]
http://imgur.com/gallery/Br00TCn

File: code/modules/research/devicecrafting/device.dm [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+/obj/item/devicecrafting/device/proc/on_add()
+	return
+
+/obj/item/devicecrafting/device/proc/on_tesla(var/power)
+	return
+
+/obj/item/devicecrafting/device/proc/on_throw(atom/target)
+	return
+
+/obj/item/devicecrafting/device/proc/on_afterattack(atom/target, mob/user, proximity_flag)
```
Comment (optimumtact) [inline code/modules/research/devicecrafting/device.dm]: why are you overriding these if they do nothing? just leave them undefined

File: tgstation.dme (hunk 4 above)
Comment (Iamgoofball) [inline tgstation.dme]: whooops

File: code/modules/research/devicecrafting/device.dm (hunk 5 above)
Comment (Iamgoofball) [inline code/modules/research/devicecrafting/device.dm]: what do you mean by overriding

it's setting up the framework for devices

**Comment by Razharas** [comment]
Explain what the fuck is this how it works and what for

**Comment by Iamgoofball** [comment]
@Razharas goofR&D

you print a device frame

you print devices

you attach trigger/effect/misc devices and they make the device do different things

**Comment by optimumtact** [comment]
how do they actually trigger actual items though, have you just not got to that point?

**Comment by Razharas** [comment]
If this is new r&d than it surely lacks the r part of it

**Comment by Iamgoofball** [comment]
@Razharas youre gonna research it by making your own devices and finding your favorites
@optimumtact that's where the hook-ins come in

File: code/modules/research/devicecrafting/obj_stuff.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/research/devicecrafting/obj_stuff.dm]: it's so I can have a structure/machine device frame and a handheld frame without snowflaking/copypaste

what the fuck we don't absolute path vars

File: code/modules/research/devicecrafting/device.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/research/devicecrafting/device.dm]: these ones are actually important so no

**Comment by LaharlMontogmmery** [comment]
>Goof reworking RnD

It BEGINS.

And I'm surprisingly OK with this.

**Comment by Shadowlight213** [comment]
Wait, so does this replace the shitcode that is assemblies?

**Comment by optimumtact** [comment]
completely different thing

**Comment by Core0verload** [comment]
\>assemblies
\>shitcode 

Did you ever saw the _old_ assemblies?

**Comment by Shadowlight213** [comment]
They were made not shit?
The old ones I guess are what I was thinking of.

**Comment by Supermichael777** [comment]
>it doesn't compile

GOOOOOOOOOF

**Comment by Bawhoppen** [comment]
branchnameaccurate

**Comment by GunHog** [comment]
Are you still working on this, Goof?

**Comment by Iamgoofball** [comment]
yeah let me finish it up with triggers

**Comment by KorPhaeron** [comment]
Stale

**Comment by GunHog** [comment]
# GOOOOOOOOOOOOOOOOOOOOOOOOOOOOOF

**Pull request closed without merging**